<div class="container">

<table style="width: 100%;"><tr>
<td>exitCheck</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Exit Error Checks</h2>

<h3>Description</h3>

<p><code>exitCheck</code> provides a mechanism to distinguish the exit status
in <code>on.exit</code> expressions.
</p>


<h3>Usage</h3>

<pre><code class="language-R">exitCheck()
</code></pre>


<h3>Details</h3>

<p>It generates a function that is used wihtin a function's body to
"flag" normal exits and in its <code>on.exit</code> expression
to check the exit status of a function.
Note that it will correctly detect errors only if all normal exit
are wrapped into a call to it.
</p>


<h3>Value</h3>

<p>Either <code>x</code> or the success status when called without arguments.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# define some function
f &lt;- function(err){

 # initialise an error checker
	success &lt;- exitCheck()

 # do something on exit that depends on the error status
	on.exit({
		if(success()) cat("Exit with no error: do nothing\n") 
		else cat("Exit with error: cleaning up the mess ...\n") 
	})
	
 # throw an error here
	if( err ) stop('There is an error')
 
	success(1+1)
}

# without error
f(FALSE)
# with error
try( f(TRUE) )

</code></pre>


</div>